### ISSUE TITLE
`Lint/LiteralAsCondition` false positive with `case`/`in` expression

### ISSUE THREAD

**Issue by tomajuskeviciute** [rubocop/rubocop#10755]
Since Ruby 2.7, `case` expressions also provide a more powerful experimental pattern matching feature via the `in` keyword:
```ruby
case { a: 1, b: 2, c: 3 }
in a: Integer => m
  "matched: #{m}"
else
  "not matched"
end
# => "matched: 1"
```

`Lint/LiteralAsCondition` offense is detected when using this new [pattern matching feature](https://docs.ruby-lang.org/en/3.0/syntax/pattern_matching_rdoc.html).

--------

## Expected behavior

No offenses detected for `Lint/LiteralAsCondition` cop.

## Actual behavior

```ruby
Offenses:

example.rb:1:6: W: Lint/LiteralAsCondition: Literal { a: 1, b: 2, c: 3 } appeared as a condition.
case { a: 1, b: 2, c: 3 }
     ^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

```ruby
case { a: 1, b: 2, c: 3 }
in a: Integer => m
  "matched: #{m}"
else
  "not matched"
end
```

## RuboCop version

```
$ bundle exec rubocop -V
1.31.0 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.0.4 x86_64-linux-musl)
  - rubocop-graphql 0.14.3
  - rubocop-performance 1.14.2
  - rubocop-rails 2.15.1
  - rubocop-rspec 2.11.1
```
